File: fore-network/src/main/java/co/early/fore/net/apollo/ErrorHandler.java
package co.early.fore.net.apollo;

import com.apollographql.apollo.api.Response;

import org.jetbrains.annotations.Nullable;


/**
 *
 * @param <F> Failure message class to be used throughout the app, such as an enum
 */
public interface ErrorHandler<F> {
    /**
     *
     * @param t throwable that caused the error, may be null
     * @param errorResponse error response from the server, the body of which may represent a custom error, may be null
     * @return the parsed error from the server
     */
    F handleError(@Nullable Throwable t, @Nullable Response errorResponse);
}

File: lib/fore-kt-android-network/src/main/java/co/early/fore/net/apollo/ErrorHandler.kt
package co.early.fore.net.apollo

import com.apollographql.apollo.api.Response

/**
 *
 * @param <F> Failure message class to be used throughout the app, such as an enum
</F> */
interface ErrorHandler<F> {
    /**
     *
     * @param t throwable that caused the error, may be null
     * @param errorResponse error response from the server, the body of which may represent a custom error, may be null
     * @return the parsed error from the server
     */
    fun handleError(t: Throwable?, errorResponse: Response<*>?): F
}